src/Components/Post/PostId.tsx

```diff
@@ -82,67 +92,154 @@ const PostId = () => {
   const onCloseClick = () => {
     setIsExpanded(false)
   }
-
-  const panelContent = (
-    <DrawerPanelContent>
-      <DrawerHead>
-        <span tabIndex={isExpanded ? 0 : -1}>
-          {error ? (
-            <TextContent>Something went wrong...</TextContent>
-          ) : (
-            <>
-              <TextContent>
-                <Text component={TextVariants.h1}>Comments</Text>
-              </TextContent>
-              <hr></hr>
-              {filteredComments.map((comment) => (
-                <Card key={comment['id']}>
-                  <CardHeader>
-                    <CardTitle>
-                      {' '}
-                      <Avatar src={avatarImg} alt="avatar" />
-                      {comment['name']}
-                    </CardTitle>
-                  </CardHeader>
-                  <CardBody>{comment['email']}</CardBody>
-                  <CardBody>{comment['body']}</CardBody>
-                </Card>
-              ))}
-            </>
-          )}
-        </span>
-        <DrawerActions>
-          <DrawerCloseButton onClick={onCloseClick} />
-        </DrawerActions>
-      </DrawerHead>
-    </DrawerPanelContent>
-  )
-
   const handleCommentSection = () => {
     setShowing(!showing)
     setComment('')
+    setEmail('')
+    setName('')
   }
 
   const addComment = () => {
-    if (comment) {
+    if (comment && name && email) {
       setComment('')
+      setEmail('')
+      setName('')
 
       setCommentData((prevState) => [
+        ...prevState,
         {
           postId: id,
           id: id,
-          name: 'Pushpa',
-          email: 'pushpa@gmail.com',
+          name: name,
+          email: email,
           body: comment,
         },
-        ...prevState,
       ])
-      console.log(commentData)
+
+      console.log(commentData.length)
     }
   }
+
+  const [name, setName] = React.useState('')
```

Comment:
Let me suggest you structure your code differently: maybe all states, then all refs, then all callback, then all effects. You can choose whatever order you want, but please let's structure the file so that it is easy to find all declarations without scrolling up and down.